File: z_cs_py_yapilanlar/CBakiye.cs
public class CBakiye
{
    CTrader Trader = null;

    ~CBakiye()
    {

    }

    public CBakiye()
    {

    }

    public CBakiye Initialize(dynamic Sistem, CTrader Trader)
    {
        this.Trader = Trader;

        return this;
    }

    public CBakiye Reset(dynamic Sistem)
    {
        return this;
    }

    public int Hesapla(dynamic Sistem, int BarIndex)
    {
        int result = 0;

        int i = BarIndex;

        // Bakiye (Puan)
        Trader.Lists.BakiyePuanList[i] = Trader.Status.BakiyePuan + Trader.Lists.KarZararPuanList[i];
        Trader.Lists.GetiriPuanList[i] = Trader.Lists.BakiyePuanList[i] - Trader.Status.IlkBakiyePuan;

        if (Trader.Flags.BakiyeGuncelle)
        {
            Trader.Status.BakiyePuan = Trader.Lists.BakiyePuanList[i];
            Trader.Status.GetiriPuan = Trader.Lists.GetiriPuanList[i];

            if (Trader.Lists.KarZararPuanList[i] >= 0)
                Trader.Status.ToplamKarPuan += Trader.Lists.KarZararPuanList[i];
            else if (Trader.Lists.KarZararPuanList[i] < 0)
                Trader.Status.ToplamZararPuan += Trader.Lists.KarZararPuanList[i];

            Trader.Status.NetKarPuan = Trader.Status.ToplamKarPuan + Trader.Status.ToplamZararPuan;
        }

        // Bakiye (Fiyat)
        Trader.Lists.BakiyeFiyatList[i] = Trader.Status.BakiyeFiyat + Trader.Lists.KarZararFiyatList[i];
        Trader.Lists.GetiriFiyatList[i] = Trader.Lists.BakiyeFiyatList[i] - Trader.Status.IlkBakiyeFiyat;

        if (Trader.Flags.BakiyeGuncelle)
        {
            Trader.Status.BakiyeFiyat = Trader.Lists.BakiyeFiyatList[i];
            Trader.Status.GetiriFiyat = Trader.Lists.GetiriFiyatList[i];

            if (Trader.Lists.KarZararFiyatList[i] >= 0)
                Trader.Status.ToplamKarFiyat += Trader.Lists.KarZararFiyatList[i];
            else if (Trader.Lists.KarZararFiyatList[i] < 0)
                Trader.Status.ToplamZararFiyat += Trader.Lists.KarZararFiyatList[i];

            Trader.Status.NetKarFiyat = Trader.Status.ToplamKarFiyat + Trader.Status.ToplamZararFiyat;
        }

        if (Trader.Status.IlkBakiyePuan != 0f)
            Trader.Lists.GetiriPuanYuzdeList[i] = 100.0f * Trader.Lists.GetiriPuanList[i] / Convert.ToSingle(Trader.Status.IlkBakiyePuan);
        else
            Trader.Lists.GetiriPuanYuzdeList[i] = 0f;

        if (Trader.Status.IlkBakiyeFiyat != 0f)
            Trader.Lists.GetiriFiyatYuzdeList[i] = 100.0f * Trader.Lists.GetiriFiyatList[i] / Convert.ToSingle(Trader.Status.IlkBakiyeFiyat);
        else
            Trader.Lists.GetiriFiyatYuzdeList[i] = 0f;

        if (Trader.Flags.BakiyeGuncelle)
        {
            Trader.Status.GetiriPuanYuzde = Trader.Lists.GetiriPuanYuzdeList[i];
            Trader.Status.GetiriFiyatYuzde = Trader.Lists.GetiriFiyatYuzdeList[i];
        }

        float k = Trader.Status.KomisyonCarpan != 0.0f ? 1.0f : 0.0f;

        Trader.Lists.GetiriFiyatNetList[i] = Trader.Lists.GetiriFiyatList[i] - Trader.Lists.KomisyonFiyatList[i] * k;
        Trader.Lists.BakiyeFiyatNetList[i] = Trader.Lists.GetiriFiyatNetList[i] + Trader.Status.IlkBakiyeFiyat;

        Trader.Lists.GetiriFiyatYuzdeNetList[i] = 0f;
        if (Trader.Status.IlkBakiyeFiyat != 0f)
            Trader.Lists.GetiriFiyatYuzdeNetList[i] = 100.0f * Trader.Lists.GetiriFiyatNetList[i] / Convert.ToSingle(Trader.Status.IlkBakiyeFiyat);

        Trader.Lists.GetiriKz[i] = Trader.Lists.GetiriFiyatList[i] / Trader.Status.VarlikAdedSayisi;
        Trader.Lists.GetiriKzNet[i] = Trader.Lists.GetiriFiyatNetList[i] / Trader.Status.VarlikAdedSayisi;

        if (i == Trader.BarCount-1)
        {
            Trader.Status.BakiyeFiyat         = Trader.Lists.BakiyeFiyatList[Trader.Lists.BakiyeFiyatList.Count-1];
            Trader.Status.GetiriFiyat         = Trader.Lists.GetiriFiyatList[Trader.Lists.GetiriFiyatList.Count-1];
            Trader.Status.GetiriKz            = Trader.Lists.GetiriKz[Trader.Lists.GetiriKz.Count-1];
            Trader.Status.GetiriFiyatYuzde    = Trader.Lists.GetiriFiyatYuzdeList[Trader.Lists.GetiriFiyatYuzdeList.Count-1];

            Trader.Status.BakiyeFiyatNet      = Trader.Lists.BakiyeFiyatNetList[Trader.Lists.BakiyeFiyatNetList.Count-1];
            Trader.Status.GetiriFiyatNet      = Trader.Lists.GetiriFiyatNetList[Trader.Lists.GetiriFiyatNetList.Count-1];
            Trader.Status.GetiriKzNet         = Trader.Lists.GetiriKzNet[Trader.Lists.GetiriKzNet.Count-1];
            Trader.Status.GetiriFiyatYuzdeNet = Trader.Lists.GetiriFiyatYuzdeNetList[Trader.Lists.GetiriFiyatYuzdeNetList.Count-1];

            Trader.Status.BakiyePuan          = Trader.Lists.BakiyePuanList[Trader.Lists.BakiyePuanList.Count-1];
            Trader.Status.GetiriPuan          = Trader.Lists.GetiriPuanList[Trader.Lists.GetiriPuanList.Count-1];
            Trader.Status.BakiyePuanNet       = Trader.Lists.BakiyePuanNetList[Trader.Lists.BakiyePuanNetList.Count-1];
            Trader.Status.GetiriPuanNet       = Trader.Lists.GetiriPuanNetList[Trader.Lists.GetiriPuanNetList.Count-1]; ;

            Trader.Status.GetiriPuanYuzdeNet  = Trader.Lists.GetiriPuanYuzdeNetList[Trader.Lists.GetiriPuanYuzdeNetList.Count-1]; ;
        }

        return result;
    }
}

File: z_cs_py_yapilanlar/CBakiye.py
class CBakiye:
    def __init__(self):
        self.Trader = None

    def __del__(self):
        pass

    def initialize(self, Sistem, Trader):
        self.Trader = Trader
        return self

    def reset(self, Sistem):
        return self

    def hesapla(self, Sistem, BarIndex):
        result = 0
        i = BarIndex

        # Bakiye (Puan)
        self.Trader.Lists.BakiyePuanList[i] = self.Trader.Status.BakiyePuan + self.Trader.Lists.KarZararPuanList[i]
        self.Trader.Lists.GetiriPuanList[i] = self.Trader.Lists.BakiyePuanList[i] - self.Trader.Status.IlkBakiyePuan

        if self.Trader.Flags.BakiyeGuncelle:
            self.Trader.Status.BakiyePuan = self.Trader.Lists.BakiyePuanList[i]
            self.Trader.Status.GetiriPuan = self.Trader.Lists.GetiriPuanList[i]

            if self.Trader.Lists.KarZararPuanList[i] >= 0:
                self.Trader.Status.ToplamKarPuan += self.Trader.Lists.KarZararPuanList[i]
            elif self.Trader.Lists.KarZararPuanList[i] < 0:
                self.Trader.Status.ToplamZararPuan += self.Trader.Lists.KarZararPuanList[i]

            self.Trader.Status.NetKarPuan = self.Trader.Status.ToplamKarPuan + self.Trader.Status.ToplamZararPuan

        # Bakiye (Fiyat)
        self.Trader.Lists.BakiyeFiyatList[i] = self.Trader.Status.BakiyeFiyat + self.Trader.Lists.KarZararFiyatList[i]
        self.Trader.Lists.GetiriFiyatList[i] = self.Trader.Lists.BakiyeFiyatList[i] - self.Trader.Status.IlkBakiyeFiyat

        if self.Trader.Flags.BakiyeGuncelle:
            self.Trader.Status.BakiyeFiyat = self.Trader.Lists.BakiyeFiyatList[i]
            self.Trader.Status.GetiriFiyat = self.Trader.Lists.GetiriFiyatList[i]

            if self.Trader.Lists.KarZararFiyatList[i] >= 0:
                self.Trader.Status.ToplamKarFiyat += self.Trader.Lists.KarZararFiyatList[i]
            elif self.Trader.Lists.KarZararFiyatList[i] < 0:
                self.Trader.Status.ToplamZararFiyat += self.Trader.Lists.KarZararFiyatList[i]

            self.Trader.Status.NetKarFiyat = self.Trader.Status.ToplamKarFiyat + self.Trader.Status.ToplamZararFiyat

        if self.Trader.Status.IlkBakiyePuan != 0.0:
            self.Trader.Lists.GetiriPuanYuzdeList[i] = 100.0 * self.Trader.Lists.GetiriPuanList[i] / float(self.Trader.Status.IlkBakiyePuan)
        else:
            self.Trader.Lists.GetiriPuanYuzdeList[i] = 0.0

        if self.Trader.Status.IlkBakiyeFiyat != 0.0:
            self.Trader.Lists.GetiriFiyatYuzdeList[i] = 100.0 * self.Trader.Lists.GetiriFiyatList[i] / float(self.Trader.Status.IlkBakiyeFiyat)
        else:
            self.Trader.Lists.GetiriFiyatYuzdeList[i] = 0.0

        if self.Trader.Flags.BakiyeGuncelle:
            self.Trader.Status.GetiriPuanYuzde = self.Trader.Lists.GetiriPuanYuzdeList[i]
            self.Trader.Status.GetiriFiyatYuzde = self.Trader.Lists.GetiriFiyatYuzdeList[i]

        k = 1.0 if self.Trader.Status.KomisyonCarpan != 0.0 else 0.0

        self.Trader.Lists.GetiriFiyatNetList[i] = self.Trader.Lists.GetiriFiyatList[i] - self.Trader.Lists.KomisyonFiyatList[i] * k
        self.Trader.Lists.BakiyeFiyatNetList[i] = self.Trader.Lists.GetiriFiyatNetList[i] + self.Trader.Status.IlkBakiyeFiyat

        self.Trader.Lists.GetiriFiyatYuzdeNetList[i] = 0.0
        if self.Trader.Status.IlkBakiyeFiyat != 0.0:
            self.Trader.Lists.GetiriFiyatYuzdeNetList[i] = 100.0 * self.Trader.Lists.GetiriFiyatNetList[i] / float(self.Trader.Status.IlkBakiyeFiyat)

        self.Trader.Lists.GetiriKz[i] = self.Trader.Lists.GetiriFiyatList[i] / self.Trader.Status.VarlikAdedSayisi
        self.Trader.Lists.GetiriKzNet[i] = self.Trader.Lists.GetiriFiyatNetList[i] / self.Trader.Status.VarlikAdedSayisi

        if i == self.Trader.BarCount - 1:
            self.Trader.Status.BakiyeFiyat = self.Trader.Lists.BakiyeFiyatList[-1]
            self.Trader.Status.GetiriFiyat = self.Trader.Lists.GetiriFiyatList[-1]
            self.Trader.Status.GetiriKz = self.Trader.Lists.GetiriKz[-1]
            self.Trader.Status.GetiriFiyatYuzde = self.Trader.Lists.GetiriFiyatYuzdeList[-1]
            self.Trader.Status.BakiyeFiyatNet = self.Trader.Lists.BakiyeFiyatNetList[-1]
            self.Trader.Status.GetiriFiyatNet = self.Trader.Lists.GetiriFiyatNetList[-1]
            self.Trader.Status.GetiriKzNet = self.Trader.Lists.GetiriKzNet[-1]
            self.Trader.Status.GetiriFiyatYuzdeNet = self.Trader.Lists.GetiriFiyatYuzdeNetList[-1]
            self.Trader.Status.BakiyePuan = self.Trader.Lists.BakiyePuanList[-1]
            self.Trader.Status.GetiriPuan = self.Trader.Lists.GetiriPuanList[-1]
            self.Trader.Status.BakiyePuanNet = self.Trader.Lists.BakiyePuanNetList[-1]
            self.Trader.Status.GetiriPuanNet = self.Trader.Lists.GetiriPuanNetList[-1]
            self.Trader.Status.GetiriPuanYuzdeNet = self.Trader.Lists.GetiriPuanYuzdeNetList[-1]

        return result
